### PR TITLE
ss-11: Add Bar Chart

### PR DESCRIPTION
**Pull request by Holovashcenko**
Add:

- vertical bar chart
- types
- constants
- styles

### CODE REVIEW

File: frontend/src/bundles/common/components/bar-chart/bar-chart.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                data: dateHelper(income),
+                backgroundColor: INCOME_COLOR,
+            },
+        ],
+    };
+
+    return <Bar options={options} data={data} className={styles.chart} />;
+};
+
+Chart.propTypes = {
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/bar-chart/bar-chart.tsx:44]: we don't use prop types.

File: frontend/src/bundles/common/components/bar-chart/bar.module.css [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.chart {
+    height: 400px;
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/bar-chart/bar.module.css:2]: I am not sure about height

File: frontend/src/bundles/common/components/bar-chart/datebar-helper.ts
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/bar-chart/datebar-helper.ts:1]: create `helpers` folder

File: frontend/src/bundles/common/components/bar-chart/bar-constants.ts
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/bar-chart/bar-constants.ts:1]: I prefer to use an enum for it

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/bar-chart/bar-chart.tsx (hunk 1 above)
Comment (Holovashcenko) [reply frontend/src/bundles/common/components/bar-chart/bar-chart.tsx]: fixed: [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/32/commits/041e22d21b1ee608d88d1097227684812d5c9000](url)

File: frontend/src/bundles/common/components/bar-chart/bar.module.css (hunk 2 above)
Comment (Holovashcenko) [reply frontend/src/bundles/common/components/bar-chart/bar.module.css]: fixed: [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/32/commits/1203288c50575a8d15e088825517d0b2f0fbb67b](url)

File: frontend/src/bundles/common/components/bar-chart/datebar-helper.ts
Comment (Holovashcenko) [reply frontend/src/bundles/common/components/bar-chart/datebar-helper.ts:1]: fixed: [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/32/commits/93188a1830f2de6114daea2b169aa626eb0fe76f](url)

File: frontend/src/bundles/common/components/bar-chart/bar-constants.ts
Comment (Holovashcenko) [reply frontend/src/bundles/common/components/bar-chart/bar-constants.ts:1]: fixed: [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/32/commits/fc005c4f325912cb59cb865fd44ee74c5a40e83d](url)

**Comment by Holovashcenko** [comment]
image
image

File: frontend/src/bundles/common/components/bar-chart/bar.enum.ts
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/bar-chart/bar.enum.ts:1]: https://www.typescriptlang.org/docs/handbook/enums.html
and
make 2 enums for it one form color, and one for label

Comment (Holovashcenko) [reply frontend/src/bundles/common/components/bar-chart/bar.enum.ts:1]: fixed: [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/32/commits/8916601af31e676ca8b9320a75a93624a4a4accf](url)

File: frontend/src/bundles/common/components/bar-chart/bar-color.enum.ts
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/bar-chart/bar-color.enum.ts:1]: create enum folder
Please check our code structure

File: frontend/src/bundles/common/components/bar-chart/bar-label.enum.ts
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/bar-chart/bar-label.enum.ts:1]: same

File: frontend/src/bundles/common/components/bar-chart/bar.type.ts
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/bar-chart/bar.type.ts:1]: Props define in the component

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/bar-chart/bar-color.enum.ts
Comment (Holovashcenko) [reply frontend/src/bundles/common/components/bar-chart/bar-color.enum.ts:1]: resolved  in  [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/32/commits/a803f521b62a6b79ebbd0a2fb0be9b2446c5946d](url)

File: frontend/src/bundles/common/components/bar-chart/bar-label.enum.ts
Comment (Holovashcenko) [reply frontend/src/bundles/common/components/bar-chart/bar-label.enum.ts:1]: resolved in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/32/commits/a803f521b62a6b79ebbd0a2fb0be9b2446c5946d](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/url)

File: frontend/src/bundles/common/components/bar-chart/bar.type.ts
Comment (Holovashcenko) [reply frontend/src/bundles/common/components/bar-chart/bar.type.ts:1]: resolved in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/32/commits/d1ac45d8abd2dfc13f95512382eb586811808e79](url)

File: frontend/src/bundles/common/components/bar-chart/bar-chart.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            {
+                label: BarLabel.OUTCOME_LABEL,
+                data: dateHelper(outcome),
+                backgroundColor: BarColor.OUTCOME_COLOR,
+            },
+            {
+                label: BarLabel.INCOME_LABEL,
+                data: dateHelper(income),
+                backgroundColor: BarColor.INCOME_COLOR,
+            },
```
Comment (VItaliy-Kalenichenko) [inline frontend/src/bundles/common/components/bar-chart/bar-chart.tsx:36]: I think it's better to replace 'income' and 'outcome' props with one datasets prop

File: frontend/src/bundles/common/components/bar-chart/config-bar.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+const options = {
```
Comment (VItaliy-Kalenichenko) [inline frontend/src/bundles/common/components/bar-chart/config-bar.ts:1]: Check if there is an interface for options

File: frontend/src/bundles/common/components/bar-chart/config-bar.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+                        argument0: string,
+                        argument1: { style: string; currency: string },
+                    ) => string;
+                }): string =>
+                    `+${value
+                        .toLocaleString('en-US', {
+                            style: 'currency',
+                            currency: 'USD',
+                        })
+                        .slice(1)} $`,
```
Comment (VItaliy-Kalenichenko) [inline frontend/src/bundles/common/components/bar-chart/config-bar.ts:16]: It's hard to understand what's going on here.
Create a separate function for the callback and interfaces to it (maybe interfaces can be imported from the package).

File: frontend/src/bundles/common/components/bar-chart/bar-chart.tsx (hunk 3 above)
Comment (Holovashcenko) [reply frontend/src/bundles/common/components/bar-chart/bar-chart.tsx]: fix in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/32/commits/81e9f408873f7c552d42647c617ef8337b7546ab](url)

File: frontend/src/bundles/common/helpers/datebar-helper/datebar-helper.ts
[the remark this answers is not in the document]
Comment (Holovashcenko) [reply frontend/src/bundles/common/helpers/datebar-helper/datebar-helper.ts:1]: fix in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/32/commits/ba1d9712cd78709371cf5c772e20edcfd2591727](url)

**Comment by Holovashcenko** [comment]
image

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**